### PR TITLE
Add extra sidecar containers config for Keda operator

### PR DESCRIPTION
**Pull request by lpetrazickisupgrade**
Add configuration parameters for specifying extra sidecar containers and extra init containers. Implements #607 

### Checklist

- [x] I have verified that my change is according to the [deprecations & breaking changes policy](https://github.com/kedacore/governance/blob/main/DEPRECATIONS.md)
- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/charts/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))
- [x] README is updated with new configuration values *(if applicable)* [learn more](https://github.com/kedacore/charts/blob/main/CONTRIBUTING.md#documentation)
- [x] A PR is opened to update KEDA core ([repo](https://github.com/kedacore/keda)) *(if applicable, ie. when deployment manifests are modified)*

Fixes #

### CODE REVIEW

File: keda/README.md [hunk 1]
```diff
@@ -67,7 +67,7 @@ their default values.
 | `certificates.certManager.duration` | string | `"8760h0m0s"` | Certificate duration |
 | `certificates.certManager.enabled` | bool | `false` | Enables Cert-manager for certificate management |
 | `certificates.certManager.generateCA` | bool | `true` | Generates a self-signed CA with Cert-manager. If generateCA is false, the secret with the CA has to be annotated with `cert-manager.io/allow-direct-injection: "true"` |
-| `certificates.certManager.issuer` | object | `{"generate":true,"group":"cert-manager.io","kind":"ClusterIssuer","name":"foo-org-ca"}` | Reference to custom Issuer. If issuer.generate is false, then issuer.group, issuer.kind and issuer.name are required |
```
Comment (JorTurFer) [inline keda/README.md:70]: Is this change intended?

**Review by JorTurFer — CHANGES_REQUESTED** [review]
Please, remove the tgz and revert the index change, we ship the chart manually aggregating all the changes that we want to ship

File: keda/README.md (hunk 1 above)
Comment (lpetrazickisupgrade) [reply keda/README.md]: Not intended. I've now copied the sentence to the values file for helm-docs

File: keda/templates/manager/deployment.yaml [hunk 2]
```diff
@@ -187,6 +187,13 @@ spec:
             {{- else }}
             {{- toYaml .Values.resources | nindent 12 }}
             {{- end }}
+        {{- if .Values.operator.extraContainers }}
+        {{- toYaml .Values.operator.extraContainers | nindent 8 }}
+        {{- end }}
+      {{- if .Values.operator.extraInitContainers }}
+      initContainers:
+      {{- toYaml .Values.operator.extraInitContainers | nindent 6 }}
```
Comment (JorTurFer) [inline keda/templates/manager/deployment.yaml:195]: is 6 the correct value? containers and initContainers are at the same level, and you've set 8 for extra containers and 6 for extra initContainers. I guess that one of them is wrong

Comment (lpetrazickisupgrade) [reply keda/templates/manager/deployment.yaml:195]: @JorTurFer Fair! There's a bit of inconsistency because the `containers:` list is indented more than the `volumes:` list. I've now set it to 8 to make it consistent with containers

**Review by JorTurFer — COMMENTED** [review]
LGTM! Thanks for the improvement! ❤️ 
Is this something that could be interesting to include in our tests? You can include them modifying the values file generated during this step: https://github.com/kedacore/charts/blob/main/.github/workflows/ci-core.yml#L90

File: keda/values.yaml [hunk 3]
```diff
@@ -83,6 +83,10 @@ operator:
     #         values:
     #         - keda-operator
     #     topologyKey: "kubernetes.io/hostname"
+  # -- Additional containers to run as part of the operator deployment
+  extraContainers: []
```
[the remark this answers is not in the document]
Comment (lpetrazickisupgrade) [reply keda/values.yaml:91]: @zroubalik Sure, examples added

**Comment by lpetrazickisupgrade** [comment]
@JorTurFer Sure, I've added it to the test run yaml

**Review by JorTurFer — APPROVED** [review]

**Comment by JorTurFer** [comment]
Thanks for your contribution!
